File: project-euler/c/p2.c
#include <stdio.h>
#include "utils.h"

int even_fib(int n) {
    int sum = 0;
    int fa = 1;
    int fb = 1;
    while (fb <= n) {
        int current = fa + fb;
        fa = fb;
        fb = current;
        if (current % 2 == 0) {
            sum += current;
        }
    }
    return sum;
}

int main() {
    int n = readint("../input2");
    printf("%d\n", even_fib(n));
}

File: project-euler/python/p2.py
def even_fib(n):
    sum = 0
    last_two = (1, 1)
    while last_two[1] <= n:
        current = last_two[0] + last_two[1]
        last_two = (last_two[1], current)
        #print(current)
        if current % 2 == 0:
            sum += current
    return sum

print(even_fib(4 * 1000 * 1000))